### PR TITLE
feat: design: On error pages (and elsewhere) offer a bug report page

### PR DESCRIPTION
**Pull request by shkeating**
# SC-1818

## Proposed changes

- adds text and buttons to add feedback to error pages
- adjusts styling of error pages to adjust accessibility issues
- adjusts styling of error pages to look a little nicer

## Reviewer notes

need to add analytics flags in where needed, and rn some of the tests are failing becuas ethey are looking for strings that dont match what i have on the page

## Setup

```
yarn dev

```

http://localhost:3000/500 [500 page]
http://localhost:3000/500dff [404 page]
---

### As the original developer, I have

- [x] Met the acceptance criteria, or will meet them in subsequent PRs or stories
  - ... 
- [x] Created new stories in Storybook if applicable
  - [ ] Checked that all Storybook accessibility checks are passing
- [x] Created/modified automated unit tests in Jest
  - [x] Including jest-axe checks when UI changes
- [x] Created/modified automated E2E tests in Cypress
  - [x] Including cypress-axe checks when UI changes
  - [x] Checked that the E2E test build is not failing
- Performed [a11y testing](https://github.com/trussworks/accessibility/blob/master/sample_a11y_testing_process.md):
  - [x] Checked responsiveness in mobile, tablet, and desktop
  - [x] Checked keyboard navigability
  - [x] Tested with [VoiceOver](https://dequeuniversity.com/screenreaders/voiceover-keyboard-shortcuts) in Safari
  - [x] Checked VO's [rotor menu](https://github.com/trussworks/accessibility/blob/master/README.md#how-to-use-the-rotor-menu) for landmarks, page heading structure and links
  - [x] Used a browser a11y tool to check for issues (WAVE, axe, ANDI or Accessibility addon tab for Storybook)
- [x] Requested a design review for user-facing changes
- For any new migrations/schema changes:
  - [x] Followed guidelines for zero-downtime deploys

### As code reviewer(s), I have

- [ ] Pulled this branch locally and tested it
- [ ] Reviewed this code and left comments
- [ ] Checked that all code is adequately covered by tests
  - [ ] Checked that the E2E test build is not failing
- [ ] Made it clear which comments need to be addressed before this work is merged
- [ ] Considered marking this as accepted even if there are small changes needed
- Performed [a11y testing](https://github.com/trussworks/accessibility/blob/master/sample_a11y_testing_process.md):
  - [ ] Checked responsiveness in mobile, tablet, and desktop
  - [ ] Checked keyboard navigability
  - [ ] Tested with [VoiceOver](https://dequeuniversity.com/screenreaders/voiceover-keyboard-shortcuts) in Safari
  - [ ] Checked VO's [rotor menu](https://github.com/trussworks/accessibility/blob/master/README.md#how-to-use-the-rotor-menu) for landmarks, page heading structure and links
  - [ ] Used a browser a11y tool to check for issues (WAVE, axe, ANDI or Accessibility addon tab for Storybook)

### As a designer reviewer, I have

- [ ] Checked in the design translated visually
- [ ] Checked behavior
- [ ] Checked different states (empty, one, some, error)
- [ ] Checked for landmarks, page heading structure, and links
- [ ] Tried to break the intended flow
- Performed [a11y testing](https://github.com/trussworks/accessibility/blob/master/sample_a11y_testing_process.md):
  - [ ] Checked responsiveness in mobile, tablet, and desktop
  - [ ] Checked keyboard navigability
  - [ ] Tested with [VoiceOver](https://dequeuniversity.com/screenreaders/voiceover-keyboard-shortcuts) in Safari
  - [ ] Checked VO's [rotor menu](https://github.com/trussworks/accessibility/blob/master/README.md#how-to-use-the-rotor-menu) for landmarks, page heading structure and links
  - [ ] Used a browser a11y tool to check for issues (WAVE, axe, ANDI or Accessibility addon tab for Storybook)

### As a test user, I have

- Run through the [Test Script](hhttps://docs.google.com/spreadsheets/d/1eV8UK0aJZ0qrzjnXf5SJ_uRiV7bpsj3yrhRFEkjOvV4/edit?usp=sharing):
  - [ ] On commercial internet in IE11
  - [ ] On commercial internet in Firefox
  - [ ] On commercial internet in Chrome
  - [ ] On commercial internet in Safari
  - [ ] On NIPR in IE11
  - [ ] On NIPR in Firefox
  - [ ] On NIPR in Chrome
  - [ ] On NIPR in Safari
  - [ ] On a mobile device in Firefox
  - [ ] On a mobile device in Chrome
  - [ ] On a mobile device in Safari
- [ ] Added any anomalous behavior to this PR

---

## Screenshots

Screen Shot 2023-04-06 at 22 36 49-fullpage
Screen Shot 2023-04-06 at 22 36 44-fullpage
Screen Shot 2023-04-06 at 22 36 40-fullpage
Screen Shot 2023-04-06 at 22 36 35-fullpage

### CODE REVIEW

**Comment by shkeating** [comment]
i messed up the jest tests will look at it in the mornin'

File: src/pages/_error.tsx [hunk 1]
```diff
@@ -20,27 +23,40 @@ const Error: NextPageWithLayout<Props> = ({ statusCode }: Props) => {
 
   return (
     <>
+      <h1>500</h1>
```
Comment (gidjin) [inline src/pages/_error.tsx:26]: Suggestion: use `statusCode` in the header if it's available.

```suggestion
      <h1>{statusCode ? statusCode : 500}</h1>
```

File: src/pages/_error.tsx [hunk 2]
```diff
@@ -20,27 +23,40 @@ const Error: NextPageWithLayout<Props> = ({ statusCode }: Props) => {
...
-            While we work on fixing that, let’s get you back to business.
+            That’s an internal server error. While we work on fixing that, let’s
+            get you back to business. You may also submit a report to us at{' '}
+            <a
+              href={`mailto:${FEEDBACK_EMAIL}?subject=${FEEDBACK_SUBJECT}`}
+              target="_blank"
+              rel="noreferrer noopener"
+              className="usa-link">
+              {FEEDBACK_EMAIL}
+            </a>{' '}
```
Comment (gidjin) [inline src/pages/_error.tsx:42]: Question: This is not a pattern I'm familiar with, is it to ensure a space in the rendered page? Would it be better as an `&nbsp;` or is that different? I'm fine with it either way.

File: src/layout/ErrorLayout/ErrorLayout.module.scss [hunk 3]
```diff
@@ -2,12 +2,12 @@
 @import 'styles/govBanner';
 @import 'styles/sfds/sfdsDependencies';
 
-$hero-image: url('/assets/images/hero.jpeg');
+$hero-image: url('/assets/images/hero-transparent.png');
```
Comment (gidjin) [inline src/layout/ErrorLayout/ErrorLayout.module.scss:5]: Question: There are a few other files that use the `hero.jpeg` including the `src/styles/_themes.scss` file. Should we update those eventually as well?

File: src/pages/404.tsx [hunk 4]
```diff
@@ -22,17 +26,23 @@ export default function Custom404() {
...
-          </LinkTo>
+          <div className="flex-align-center">
+            <LinkTo className="usa-button usa-button--secondary" href="/">
+              Take me home
+            </LinkTo>
+            <a
+              href={`mailto:${FEEDBACK_EMAIL}?subject=${FEEDBACK_SUBJECT}`}
+              target="_blank"
+              rel="noreferrer noopener"
+              className="usa-button usa-button--outline usa-button--inverse">
```
Comment (gidjin) [inline src/pages/404.tsx:40]: Question: These links don't have a `hover` state, is that on purpose?

**Review by gidjin — COMMENTED** [review]
Changes look good to me. I added some tests for you, so I'll wait for @abbyoung or @jcbcapps to approve this change :)

File: src/pages/_error.tsx (hunk 2 above)
Comment (shkeating) [reply src/pages/_error.tsx:44]: I copied it over from the feedback card component, i do not have strong opinions about it

File: src/layout/ErrorLayout/ErrorLayout.module.scss (hunk 3 above)
Comment (shkeating) [reply src/layout/ErrorLayout/ErrorLayout.module.scss:5]: those other components probably don't want a transparent image, so probably not

File: src/pages/404.tsx (hunk 4 above)
Comment (shkeating) [reply src/pages/404.tsx:40]: no it is not

Comment (shkeating) [reply src/pages/404.tsx:40]: they do have a hover state on my end? the outline button goes from cosmic-latte-light to white, and the secondary button gets a dark background with white text

File: src/layout/ErrorLayout/ErrorLayout.module.scss [hunk 5]
```diff
@@ -2,12 +2,12 @@
...
 
 .errorContainer {
   display: flex;
   flex-direction: column;
-  min-height: 100vh;
```
Comment (jcbcapps) [inline src/layout/ErrorLayout/ErrorLayout.module.scss:10]: Was removing this intentional? On a larger screen I'm seeing a big gap at the bottom of the page.
Screenshot 2023-04-10 at 11 25 55 AM

**Review by jcbcapps — APPROVED** [review]
LGTM aside from the CSS question I asked, but I don't think that is blocking. Also, when running tests locally, I noticed test coverage was pretty low so I added a test for the `_error` page.

File: src/layout/ErrorLayout/ErrorLayout.module.scss (hunk 5 above)
Comment (shkeating) [reply src/layout/ErrorLayout/ErrorLayout.module.scss:10]: oop i guess it was doing something after all, good catch lol. im going to put that back

**Review by gidjin — APPROVED** [review]

**Review by malworks — APPROVED** [review]

**Pull request merged**